nym-vpn-lib/src/platform/android.rs

```diff
@@ -105,23 +104,31 @@ pub extern "system" fn Java_net_nymtech_vpn_NymVpnService_initVPN(
     LOAD_CLASSES.call_once(|| env.preload_classes(CLASSES.iter().cloned()));
 
     let context = AndroidContext {
-        jvm: Arc::new(env.get_java_vm().expect("Get JVM instance")),
+        jvm: Arc::new(
+            env.get_java_vm()
+                .map_err(|e| error!("Could not get jvm {:?}", e))
+                .expect("Could not get jvm"),
+        ),
         vpn_service: env
             .new_global_ref(vpn_service)
-            .expect("Create global reference"),
+            .map_err(|e| error!("Could not get vpn_service {:?}", e))
+            .expect("Could not get vpn_service"),
     };
-    let api_url = Url::from_str(&String::from_java(&env, api_url)).expect("Invalid api url");
-    let explorer_url =
-        Url::from_str(&String::from_java(&env, explorer_url)).expect("Invalid explorer url");
     let entry_gateway: EntryPoint = serde_json::from_str(&String::from_java(&env, entry_gateway))
-        .expect("Invalid entry gateway");
-    let exit_router: ExitPoint =
-        serde_json::from_str(&String::from_java(&env, exit_router)).expect("Invalid exit router");
+        .map_err(|e| error!("Could not parse entry point {:?}", e))
+        .unwrap();
```

Comment:
Do we want to panic on incorrect input? Earlier in the function I see a naked `return;`, consider doing the same unless it's possible to return an error to the caller